base/path.jl

```diff
@@ -125,6 +125,23 @@ abspath(a::AbstractString, b::AbstractString...) = abspath(joinpath(a,b...))
     end
 end
 
+@windows_only longpath(path::AbstractString) = longpath(utf16(path))
+@windows_only function longpath(path::UTF16String)
+    p = UInt32((sizeof(path)>>2) + 1)
```

Comment:
Why are you dividing by 4 (`>>2`) instead of by 2 (`>>1`)?  Presumably what you want is the number of code units, including the null terminator.   Maybe just `UInt32(length(path.data))`?